serve/mlc_serve/model/model_common.py

```diff
@@ -75,13 +75,22 @@ def _is_safe_to_sample(prob_like):
     logits = torch.from_dlpack(logits)
     num_seq = len(sampling_params)
 
-    mask_random = torch.tensor(
+    mask_random_cpu = torch.tensor(
         [p.sampling_type == SamplingType.RANDOM for p in sampling_params],
-        dtype=torch.bool,
+        dtype=torch.bool
     )
-    mask_greedy = torch.logical_not(mask_random)
+    mask_greedy_cpu = torch.logical_not(mask_random_cpu)
+    if logits.device == torch.device("cpu"):
+        mask_random_dvc = mask_random_cpu
+        mask_greedy_dvc = mask_greedy_cpu
+    else:  # gpu
+        mask_random_dvc = torch.tensor(
```

Comment:
`mask_random_gpu = `mask_random_cpu.to("cuda")`

Reply:
You are still creating the CPU mask twice otherwise (PT will create another one when it creates a GPU mask).

Reply:
Oh yes, thanks!